File: THCS/2023_2024/HaiDuong_2324/Cau3/Cau3.py
def is_prime(num):
    """Kiểm tra số nguyên tố."""
    if num < 2:
        return False  # Số nhỏ hơn 2 không phải số nguyên tố
    if num == 2:
        return True   # 2 là số nguyên tố
    if num % 2 == 0:
        return False  # Các số chẵn lớn hơn 2 không phải số nguyên tố

    for i in range(3, int(num**0.5) + 1, 2):  # Kiểm tra các số lẻ từ 3 đến sqrt(num)
        if num % i == 0:
            return False  # Nếu num chia hết cho i, num không phải số nguyên tố
    return True  # num là số nguyên tố

def find_smallest_prime(n):
    """Tìm số nguyên tố nhỏ nhất lớn hơn hoặc bằng n."""
    while not is_prime(n):
        n += 1  # Tăng n lên cho đến khi tìm được số nguyên tố
    return n

if __name__ == "__main__":
    n = int(input())  # Nhập số nguyên dương n
    result = find_smallest_prime(n)
    print(result)  # In ra kết quả

File: THCS/2023_2024/HaiDuong_2324/Cau3/Cau3.cpp
#include <iostream>
using namespace std;

// Hàm kiểm tra số nguyên tố
bool isPrime(int num) {
    if (num < 2) return false; // Số nhỏ hơn 2 không phải số nguyên tố
    if (num == 2) return true; // 2 là số nguyên tố
    if (num % 2 == 0) return false; // Các số chẵn lớn hơn 2 không phải số nguyên tố

    for (int i = 3; i * i <= num; i += 2) { // Kiểm tra các số lẻ từ 3 đến sqrt(num)
        if (num % i == 0) {
            return false; // Nếu num chia hết cho i, num không phải số nguyên tố
        }
    }
    return true; // num là số nguyên tố
}

int main() {
    // Nhập số nguyên dương n
    int n;
    cin >> n;

    // Tìm số nguyên tố nhỏ nhất lớn hơn hoặc bằng n
    while (!isPrime(n)) {
        n++; // Tăng n lên cho đến khi tìm được số nguyên tố
    }

    // In ra kết quả
    cout << n << endl;

    return 0;
}
